samples: net: ocpp: Fix cast warning

seankyer opened the pull request:
Fix warnings generated for casting between pointers and integers of different sizes in ocpp sample.

Before patch:
```
~/zephyrproject/zephyr$ west twister -p native_sim/native/64 -T samples/net/ocpp/
Renaming previous output directory to /home/skyer/zephyrproject/zephyr/twister-out.9
INFO    - Using Ninja..
INFO    - Zephyr version: v4.2.0-1599-g6558eca56cd3
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testsuite list...
INFO    - Writing JSON report /home/skyer/zephyrproject/zephyr/twister-out/testplan.json
INFO    - JOBS: 32
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
ERROR   - native_sim/native/64      sample.net.ocpp                                    ERROR: Build failure
ERROR   - see: /home/skyer/zephyrproject/zephyr/twister-out/native_sim_native_64/host/samples/net/ocpp/sample.net.ocpp/build.log
INFO    - Total complete:    1/   1  100%  built (not run):    0, filtered:    0, failed:    0, error:    1
INFO    - 1 test scenarios (1 configurations) selected, 0 configurations filtered (0 by static filter, 0 at runtime).
INFO    - 0 of 1 executed test configurations passed (0.00%), 0 built (not run), 0 failed, 1 errored, with no warnings in 5.19 seconds.
INFO    - 0 of 1 executed test cases passed (0.00%), 1 blocked on 1 out of total 1139 platforms (0.09%).
INFO    - 0 test configurations executed on platforms, 1 test configurations were only built.
INFO    - Saving reports...
INFO    - Writing JSON report /home/skyer/zephyrproject/zephyr/twister-out/twister.json
INFO    - Writing xunit report /home/skyer/zephyrproject/zephyr/twister-out/twister.xml...
INFO    - Writing xunit report /home/skyer/zephyrproject/zephyr/twister-out/twister_report.xml...
INFO    - -+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
INFO    - The following issues were found (showing the top 10 items):
INFO    - 1) sample.net.ocpp on native_sim/native/64 error (Build failure - error: cast from pointer to integer of different size [-Werror=pointer-to-int-cast])
INFO    - 
INFO    - To rerun the tests, call twister using the following commandline:
INFO    - west twister -p <PLATFORM> -s <TEST ID>, for example:
INFO    - 
INFO    - west twister -p native_sim/native/64 -s sample.net.ocpp
INFO    - or with west:
INFO    - west build -p -b native_sim/native/64 samples/net/ocpp -T sample.net.ocpp
INFO    - -+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
INFO    - Run completed
```

With patch:
```
~/zephyrproject/zephyr$ west twister -p native_sim/native/64 -T samples/net/ocpp/
Renaming previous output directory to /home/skyer/zephyrproject/zephyr/twister-out.8
INFO    - Using Ninja..
INFO    - Zephyr version: v4.2.0-1600-ga75ea8f63059
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testsuite list...
INFO    - Writing JSON report /home/skyer/zephyrproject/zephyr/twister-out/testplan.json
INFO    - JOBS: 32
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - Total complete:    1/   1  100%  built (not run):    1, filtered:    0, failed:    0, error:    0
INFO    - 1 test scenarios (1 configurations) selected, 0 configurations filtered (0 by static filter, 0 at runtime).
INFO    - 0 of 1 executed test configurations passed (0.00%), 1 built (not run), 0 failed, 0 errored, with no warnings in 8.02 seconds.
INFO    - 0 of 0 executed test cases passed (0.00%) on 0 out of total 1139 platforms (0.00%).
INFO    - 1 selected test cases not executed: 1 not run (built only).
INFO    - 0 test configurations executed on platforms, 1 test configurations were only built.
INFO    - Saving reports...
INFO    - Writing JSON report /home/skyer/zephyrproject/zephyr/twister-out/twister.json
INFO    - Writing xunit report /home/skyer/zephyrproject/zephyr/twister-out/twister.xml...
INFO    - Writing xunit report /home/skyer/zephyrproject/zephyr/twister-out/twister_report.xml...
INFO    - Run completed
```

Addresses Twister failure seen in: https://github.com/zephyrproject-rtos/zephyr/actions/runs/16948699457/job/48036432995?pr=93461#step:13:1975 for the `native_sim/native/64` target.

seankyer commented:
This issue is blocking https://github.com/zephyrproject-rtos/zephyr/pull/93461

ssekar15 reviewed (changes requested):
Thanks for your effort, please split sample and subsys changes into two commit.

ssekar15 commented:
> This issue is blocking #93461

@kartben not sure how missed by CI on actual PR

seankyer committed:
net: ocpp: Fix cast warning

Fix warnings generated from casting pointers
of different sizes and string literals in ocpp

Signed-off-by: Sean Kyer <Sean.Kyer@analog.com>

seankyer committed:
samples: net: ocpp: Fix cast warning

Fix warnings generated for casting between pointers
and integers of different sizes in ocpp sample.

Signed-off-by: Sean Kyer <Sean.Kyer@analog.com>

seankyer commented:
> Thanks for your effort, please split sample and subsys changes into two commit.

no problem. Just pushed with two seperate commits.

seankyer commented:
@kartben could you take a look when you get a moment? Thank you. This is blocking progress on another PR.
